src/app/pages/movie-details/movie-details.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component } from '@angular/core';
+import { DetailedDataAPIService } from '../../services/detailed-data-api.service';
+import { ActivatedRoute } from '@angular/router';
+import { BehaviorSubject, switchMap, tap } from 'rxjs';
+import { crewFilterFunc } from 'src/app/components/detailed-page/helpers/crew-filter-func';
+@Component({
+  selector: 'mdb-movie-details',
+  templateUrl: './movie-details.component.html',
+})
+export class MovieDetailsComponent {
+  constructor(
+    private route: ActivatedRoute,
+    private detailedDataService: DetailedDataAPIService
+  ) {}
+
+  public castToShow = 10;
+
+  public isLoading$ = new BehaviorSubject<boolean>(true);
+
+  public crewFilterFunc = crewFilterFunc([
+    'director',
```

Comment:
Probably better to use Enum

Reply:
Added enum of main movie crew job roles.